File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        Stack<Character> stack = new Stack<>();
        
        for(int i=0; i<s.length(); i++) {
            if(s.charAt(i) == '(') {
                stack.push('(');
            }
            else {
                if(stack.size() > 0) {
                    stack.pop();
                } else {
                    return false;
                }
            }
        }
        
        if(stack.size() != 0) return false;
        else return true;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    
    const stack = [];
    
    for(let bracket of s) {
        if(bracket === "(") {
            stack.push(1);
        } else if(stack.length > 0) {
            stack.pop();
        } else {
            return false
        }
    }
    
    if(stack.length !== 0) {
        return false
    }

    return true
}